netdata-installer.sh

```diff
@@ -1022,6 +1027,175 @@ install_go() {
+    echo >&2 " Using eBPF Kernel Package built against Linux 4.14"
+    kpkg="4_14"
+  else
+    echo >&2 " ERROR: Cannot detect a valid libc on your system!"
+    return 1
+  fi
+
+  echo "${kpkg}"
+  return 0
+}
+
+should_install_ebpf() {
+  if [ "${NETDATA_ENABLE_EBPF:=0}" -ne 1 ]; then
+    run_failed "ebpf not enabled. --enable-ebpf to enable"
+    return 1
+  fi
+
+  # Get and Parse Kernel Version
+  kver="$(get_kernel_version)"
+  kver="${kver:-0}"
+
+  # Check Kernel Compatibility
+  if ! get_compatible_kernel_for_ebpf "${kver}" > /dev/null; then
+    echo >&2 " Detected Kernel: ${kver}"
+    run_failed "Kernel incompatible. Please contact NetData support!"
+    return 1
+  fi
+
+  # Check Kernel Config
+
+  tmp="$(mktemp -t -d netdata-ebpf-XXXXXX)"
+
+  echo >&2 " Downloading check-kernel-config.sh ..."
+  if ! get "https://raw.githubusercontent.com/netdata/kernel-collector/master/tools/check-kernel-config.sh" > "${tmp}"/check-kernel-config.sh; then
+    run_failed "Failed to download check-kernel-config.sh"
+    echo 2>&" Removing temporary directory ${tmp} ..."
+    rm -rf "${tmp}"
+    return 1
+  fi
+
+  run chmod +x "${tmp}"/check-kernel-config.sh
+
+  if ! run "${tmp}"/check-kernel-config.sh; then
+    run_failed "Kernel unsupported or missing required config"
+    return 1
+  fi
+
+  rm -rf "${tmp}"
+
+  # TODO: Check for current vs. latest version
+
+  return 0
+}
+
+install_ebpf() {
+  if ! should_install_ebpf; then
+    return 0
+  fi
+
+  progress "Installing eBPF plugin"
+
+  # Get and Parse Kernel Version
+  kver="$(get_kernel_version)"
+  kver="${kver:-0}"
+
+  # Get Compatible eBPF Kernel Package
+  kpkg="$(get_compatible_kernel_for_ebpf "${kver}")"
+
+  # Detect libc
+  libc="$(detect_libc)"
+
+  PACKAGE_TARBALL="netdata_ebpf-${kpkg}-${libc}.tar.xz"
+
+  echo >&2 " Getting latest eBPF Package URL for ${PACKAGE_TARBALL} ..."
+
+  PACKAGE_TARBALL_URL=
+  PACKAGE_TARBALL_URL="$(get "https://api.github.com/repos/netdata/kernel-collector/releases/latest" | grep -o -E "\"browser_download_url\": \".*${PACKAGE_TARBALL}\"" | sed -e 's/"browser_download_url": "\(.*\)"/\1/')"
+
+  if [ -z "${PACKAGE_TARBALL_URL}" ]; then
+    run_failed "Could not get latest eBPF Package URL for ${PACKAGE_TARBALL}"
+    return 1
+  fi
+
+  tmp="$(mktemp -t -d netdata-ebpf-XXXXXX)"
+
+  echo >&2 " Downloading eBPF Package ${PACKAGE_TARBALL_URL} ..."
+  if ! get "${PACKAGE_TARBALL_URL}" > "${tmp}"/"${PACKAGE_TARBALL}"; then
+    run_failed "Failed to download latest eBPF Package ${PACKAGE_TARBALL_URL}"
+    echo 2>&" Removing temporary directory ${tmp} ..."
+    rm -rf "${tmp}"
+    return 1
+  fi
+
+  echo >&2 " Extracting ${PACKAGE_TARBALL} ..."
+  tar -xf "${tmp}/${PACKAGE_TARBALL}" -C "${tmp}"
+
+  echo >&2 " Finding suitable lib directory ..."
+  libdir=
+  libdir="$(ldconfig -v 2> /dev/null | grep ':$' | sed -e 's/://' | sort -r | grep 'usr' | head -n 1)"
+  if [ -z "${libdir}" ]; then
```

Comment:
@prologic I added these lines to fix problems with Fedora.

Reply:
Fedora 31 now builds fine

Reply:
Thank you to confirm! :)

Reply:
Nice

Reply:
Awesome :)